crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -840,4 +983,263 @@ b = z
         assert_public_ty(&db, "src/a.py", "x", "Literal[1] | None");
         Ok(())
     }
+
+    #[test]
+    fn simple_if() -> anyhow::Result<()> {
+        let mut db = setup_db();
+
+        db.write_dedented(
+            "src/a.py",
+            "
+                y = 1
+                y = 2
+                if flag:
+                    y = 3
+                x = y
```

Comment:
Do you intentionally use an extra indent for the inline code? 
```suggestion
            y = 1
            y = 2
            if flag:
                y = 3
            x = y
```